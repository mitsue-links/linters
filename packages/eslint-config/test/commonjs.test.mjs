import path from 'path';
import {lintFile} from './helper.mjs';

test('commonjs_ok.cjs', async () => {
    const filePath = 'commonjs_ok.cjs';
    const data = await lintFile(path.join('fixtures', filePath));

    expect(data).toStrictEqual([{
        filePath,
        messages: [],
    }]);
});

test('commonjs_ok.mjs', async () => {
    const filePath = 'commonjs_ok.mjs';
    const data = await lintFile(path.join('fixtures', filePath));

    expect(data).toMatchSnapshot();
});


test('commonjs_error.cjs', async () => {
    const filePath = 'commonjs_error.cjs';
    const data = await lintFile(path.join('fixtures', filePath));

    expect(data).toMatchSnapshot();
});

test('commonjs_error.mjs', async () => {
    const filePath = 'commonjs_error.mjs';
    const data = await lintFile(path.join('fixtures', filePath));

    expect(data).toMatchSnapshot();
});
