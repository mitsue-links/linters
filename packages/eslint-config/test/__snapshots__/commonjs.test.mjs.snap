// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commonjs_error.cjs 1`] = `
[
  {
    "filePath": "commonjs_error.cjs",
    "messages": [
      {
        "line": 3,
        "ruleId": "no-restricted-syntax",
        "severity": 2,
      },
      {
        "line": 3,
        "ruleId": "no-with",
        "severity": 2,
      },
      {
        "line": 4,
        "ruleId": "no-loss-of-precision",
        "severity": 1,
      },
      {
        "line": 4,
        "ruleId": "no-octal",
        "severity": 2,
      },
      {
        "line": 5,
        "ruleId": "no-octal",
        "severity": 2,
      },
      {
        "line": 7,
        "ruleId": "no-octal-escape",
        "severity": 2,
      },
      {
        "line": 8,
        "ruleId": "no-octal-escape",
        "severity": 2,
      },
      {
        "line": 10,
        "ruleId": "no-nonoctal-decimal-escape",
        "severity": 2,
      },
      {
        "line": 11,
        "ruleId": "no-nonoctal-decimal-escape",
        "severity": 2,
      },
    ],
  },
]
`;

exports[`commonjs_error.mjs 1`] = `
[
  {
    "filePath": "commonjs_error.mjs",
    "messages": [
      {
        "line": 3,
        "ruleId": "Parsing error: 'with' in strict mode",
        "severity": 2,
      },
    ],
  },
]
`;

exports[`commonjs_ok.mjs 1`] = `
[
  {
    "filePath": "commonjs_ok.mjs",
    "messages": [
      {
        "line": 4,
        "ruleId": "strict",
        "severity": 2,
      },
    ],
  },
]
`;
